async-nats/src/jetstream/mod.rs

```diff
@@ -138,3 +139,67 @@ impl From<Message> for crate::Message {
         source.message
     }
 }
+
+impl Message {
+    pub async fn ack(&self) -> Result<(), Error> {
+        if let Some(ref reply) = self.reply {
+            self.context
+                .client
+                .publish(reply.to_string(), "".into())
+                .await
+        } else {
+            Err(Box::new(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "No reply subject, not a JetStream message",
+            )))
+        }
+    }
+
+    pub async fn ack_kind(&self, kind: AckKind) -> Result<(), Error> {
```

Comment:
Bikeshed (verb over noun):

```suggestion
    pub async fn ack_with(&self, kind: AckKind) -> Result<(), Error> {
```